openmp/runtime/cmake/LibompHandleFlags.cmake

```diff
@@ -140,7 +140,7 @@ function(libomp_get_libflags libflags)
   if(LIBOMP_HAVE_SHM_OPEN_WITH_LRT)
     libomp_append(libflags_local -lrt)
   endif()
-  if(${CMAKE_SYSTEM_NAME} MATCHES "DragonFly|FreeBSD")
```

Comment:
OS lists like this are bad, can you use check_library_exists

Reply:
I think `check_library_exists` is not for the purpose of this case. For example, `libm` exists on a lot of systems, but we only want to link against it with proper linker flag when OS is BSD.

Reply:
but why, you can do any kind of try_compile test